src/main/scala/Main.scala

```diff
@@ -0,0 +1,50 @@
+/* SimpleApp.scala */
+import org.apache.spark.sql.SparkSession
+import org.apache.log4j.Logger
+import org.apache.log4j.Level
+
+import shingler.Shingler
+import hashing.Hasher
+import hashing.MinHasher
+import comparator.Comparator
+
+object Main {
```

Comment:
Scala has an App extension that can be added to an object and it takes care of the "def main..." in a cleaner way